app/src/main/res/values-ar/strings.xml

```diff
@@ -13,11 +14,134 @@
+    <string name="unit_volume_tablespoon_imperial">Tablespoon (Imperial)</string>
+    <string name="unit_mass">Mass</string>
+    <string name="unit_mass_gram">Gram</string>
+    <string name="unit_mass_kilogram">Kilogram</string>
+    <string name="unit_mass_milligram">Milligram</string>
+    <string name="unit_mass_microgram">Microgram</string>
+    <string name="unit_mass_tonne">Tonne</string>
+    <string name="unit_mass_pound">Pound</string>
+    <string name="unit_mass_ounce">Ounce</string>
+    <string name="unit_mass_grain">Grain</string>
+    <string name="unit_mass_dram">Dram</string>
+    <string name="unit_mass_stone">Stone</string>
+    <string name="unit_mass_long_ton">Long Ton</string>
+    <string name="unit_mass_short_ton">Short Ton</string>
+    <string name="unit_mass_carat">Carat</string>
+    <string name="unit_mass_carat_metric">Carat (Metric)</string>
+    <string name="unit_temperature">Temperature</string>
+    <string name="unit_temperature_celsius">Celsius</string>
+    <string name="unit_temperature_delisle">Delisle</string>
+    <string name="unit_temperature_fahrenheit">Fahrenheit</string>
+    <string name="unit_temperature_newton">Newton</string>
+    <string name="unit_temperature_rankine">Rankine</string>
+    <string name="unit_temperature_reaumur">Réaumur</string>
+    <string name="unit_temperature_romer">Rømer</string>
+    <string name="unit_temperature_gas_mark">Regulo Gas Mark</string>
+    <string name="unit_temperature_kelvin">Kelvin</string>
+    <string name="unit_time">Time</string>
+    <string name="unit_time_hour">Hour</string>
+    <string name="unit_time_minute">Minute</string>
+    <string name="unit_time_second">Second</string>
+    <string name="unit_time_millisecond">Millisecond</string>
+    <string name="unit_time_microsecond">Microsecond</string>
+    <string name="unit_time_nanosecond">Nanosecond</string>
+    <string name="unit_time_picosecond">Picosecond</string>
+    <string name="unit_time_day">Day</string>
+    <string name="unit_time_week">Week</string>
+    <string name="unit_time_month_full">Month (Full)</string>
+    <string name="unit_time_month_hollow">Month (Hollow)</string>
+    <string name="unit_time_month_synodic">Month (Synodic)</string>
+    <string name="unit_time_month_gregorian_average">Month (Gregorian average)</string>
+    <string name="unit_time_year_leap">Year (Leap)</string>
+    <string name="unit_time_year_gregorian">Year (Gregorian)</string>
+    <string name="unit_time_year_julian">Year (Julian)</string>
+    <string name="unit_time_jiffy">Jiffy</string>
+    <string name="unit_time_moment">Moment</string>
+    <string name="unit_time_fortnight">Fortnight</string>
+    <string name="unit_time_decade">Decade</string>
+    <string name="unit_time_century">Century</string>
+    <string name="unit_time_millennium">Millennium</string>
+    <string name="unit_time_atomic_unit">Atomic Unit</string>
```

Comment:
we have way too many units everywhere, try removing the less popular ones please. Use the sample app for inspiration, no need to go this advanced mode.